File: 0118-pascals-triangle/0118-pascals-triangle.java
class Solution {
    public List<List<Integer>> generate(int numRows) {
        List<List<Integer>> result = new ArrayList<>();

        if (numRows >= 1) {
            result.add(Arrays.asList(1)); 
        }

        for (int i = 1; i < numRows; i++) {
            List<Integer> answer = new ArrayList<>();
            answer.add(1);

            List<Integer> previousRow = result.get(i - 1);
            for (int j = 1; j < i; j++) {
                int sum = previousRow.get(j - 1) + previousRow.get(j);
                answer.add(sum);
            }
            
            answer.add(1);
            result.add(answer);
        }

        return result;
    }
}

File: 0118-pascals-triangle/0118-pascals-triangle.py
class Solution(object):
    def generate(self, numRows):
        """
        :type numRows: int
        :rtype: List[List[int]]
        """
        
        res = [[1]]

        if numRows > 1:
            for i in range(1, numRows):
                prev = [0] + res[i-1] + [0]
                row = [prev[j] + prev[j+1] for j in range(len(prev) - 1)]
                res.append(row)

        return res

        